extensions/http/src/main/java/org/mule/extension/http/internal/listener/grizzly/GrizzlyServerManager.java

```diff
@@ -119,34 +119,30 @@ private void configureServerSocketProperties(TCPNIOTransportBuilder transportBui
         {
             transportBuilder.setLinger(serverSocketProperties.getLinger());
         }
-        if (serverSocketProperties.getReuseAddress() != null)
-        {
-            transportBuilder.setReuseAddress(serverSocketProperties.getReuseAddress());
-        }
-        if (serverSocketProperties.getSendTcpNoDelay() != null)
-        {
-            transportBuilder.setTcpNoDelay(serverSocketProperties.getSendTcpNoDelay());
-        }
-        if (serverSocketProperties.getReceiveBacklog() != null)
-        {
-            transportBuilder.setServerConnectionBackLog(serverSocketProperties.getReceiveBacklog());
-        }
+
         if (serverSocketProperties.getReceiveBufferSize() != null)
         {
             transportBuilder.setReadBufferSize(serverSocketProperties.getReceiveBufferSize());
         }
+
         if (serverSocketProperties.getSendBufferSize() != null)
         {
             transportBuilder.setWriteBufferSize(serverSocketProperties.getSendBufferSize());
         }
-        if (serverSocketProperties.getServerTimeout() != null)
+
+        if (serverSocketProperties.getClientTimeout() != null)
         {
-            transportBuilder.setServerSocketSoTimeout(serverSocketProperties.getServerTimeout());
+            transportBuilder.setClientSocketSoTimeout(serverSocketProperties.getClientTimeout());
         }
-        if (serverSocketProperties.getTimeout() != null)
+
+        if (serverSocketProperties.getServerTimeout() != null)
         {
-            transportBuilder.setClientSocketSoTimeout(serverSocketProperties.getTimeout());
+            transportBuilder.setServerSocketSoTimeout(serverSocketProperties.getServerTimeout());
         }
+
+        transportBuilder.setReuseAddress(serverSocketProperties.getReuseAddress());
```

Comment:
Those 3 fields (reuseAddress, connectionTimeout and sendTcpNoDelay) are no longer being represented with boxed types. We now use native types with some established default value.